src/command_modules/azure-cli-resource/tests/test_resource_validators.py

```diff
@@ -18,6 +22,94 @@ def setUp(self):
+            },
+            {
+                'test': 'name and group',
+                'name': 'foo',
+                'resource_group_name': 'foo',
+            },
+            {
+                'test': 'name, group, type [compact]',
+                'name': 'foo',
+                'resource_group_name': 'bar',
+                'resource_name': 'baz',
+                'resource_type': 'Microsoft.Compute/VirtualMachines'
+            },
+            {
+                'test': 'name, group, type, namespace',
+                'name': 'foo',
+                'resource_group_name': 'bar',
+                'resource_name': 'baz',
+                'resource_type': 'VirtualMachines',
+                'resource_provider_namespace': 'Microsoft.Compute',
+            },
+            {
+                'test': 'name, group, type, namespace, parent',
+                'name': 'foo',
+                'resource_group_name': 'bar',
+                'resource_name': 'baz',
+                'resource_type': 'VirtualMachines',
+                'resource_provider_namespace': 'Microsoft.Compute',
+                'parent_resource_path': 'Foo.Bar/baz',
+            }
+        ]
+        for valid_namespace in valid:
+            try:
+                # If unexpected invalid, this throws, so no need for asserts
+                validate_lock_parameters(valid_namespace)
+            except CLIError as ex:
+                self.fail('Test {} failed. {}'.format(valid_namespace['test'], ex))
+    
+    
+    def test_validate_lock_params_invalid(self):
+        invalid = [
+            {
+                'test': 'just name and type',
+                'name': 'foo',
+                'resource_type': 'baz'
+            },
+            {
+                'test': 'name and group and type',
+                'name': 'foo',
+                'resource_group_name': 'foo',
+                'resource_type': 'bar',
+            },
+            {
+                'test': 'name, group, no type',
+                'name': 'foo',
+                'resource_group_name': 'bar',
+                'resource_name': 'baz',
+            },
+            {
+                'test': 'name, group, type, namespace',
+                'name': 'foo',
+                'resource_group_name': 'bar',
+                'resource_name': 'baz',
+                'resource_provider_namespace': 'Microsoft.Compute',
+            },
+            {
+                'test': 'name, group, type, namespace, parent',
+                'name': 'foo',
+                'resource_group_name': 'bar',
+                'resource_type': 'VirtualMachines',
+                'resource_provider_namespace': 'Microsoft.Compute',
+                'parent_resource_path': 'Foo.Bar/baz',
+            }
+        ]
+        for invalid_namespace in invalid:
+            try:
+                validate_lock_parameters(invalid_namespace)
+            except CLIError as ex:
+                # expected
+                continue
```

Comment:
You can use the `assertRaise`: https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises